griptape/utils/chat.py

```diff
@@ -16,22 +17,39 @@ class Chat:
     prompt_prefix: str = field(default="Q: ", kw_only=True)
     response_prefix: str = field(default="A: ", kw_only=True)
     output_fn: Callable[[str], None] = field(
-        default=Factory(lambda: print), kw_only=True
+        default=lambda x: print(x, end=""), kw_only=True
     )
+    stream: bool = field(default=False, kw_only=True)
+
+    def start(self, initial_turn: str = None) -> None:
+        if initial_turn:
+            self.output_fn(initial_turn + "\n")
```

Comment:
What is the reason for adding a newline? And if we're going to keep it, let's move it to line 20 inside of the default.

Reply:
I changed the default output_fn to print(x, end="") to suppress the newline after each chunk returned during streaming. Utility needs both streamed and non-streamed output_fn. Having only one output_fn and adding \n or composing it to suppress the \n when looping over the generator feels very 'print' centric. 

Would having output_fn and stream_output_fn be the best solution?

Reply:
Yeah I think two functions may be our best best here. Let's try it out and see how it looks.